projects/lib/src/board/westernboard.cpp

```diff
@@ -631,17 +655,32 @@ bool WesternBoard::vSetFenString(const QStringList& fen)
 	m_sign = (side == Side::White) ? 1 : -1;
 	if (*token != "-")
 	{
-		setEnpassantSquare(squareIndex(*token));
+		int epSq = squareIndex(*token);
+		setEnpassantSquare(epSq);
 		if (m_enpassantSquare == 0)
 			return false;
 
+		Piece ownPawn(side, Pawn);
+		Piece opPawn(side.opposite(), Pawn);
+		int matchesOwn = 0;
+		int epTgt = 0;
+
+		for ( PawnStep pStep: pawnSteps )
```

Comment:
It's almost always possible to avoid deep copies, but with Qt containers not always obvious. The "Goodbye Q_FOREACH" blog post is a good guide on this. Because Qt containers use shared data, this does NOT result in a deep copy:

`const auto list = pawnSteps;`

But now you can safely iterate over `list` without an accidental deep copy, because the list is `const`.